eventEmitter.js

```diff
@@ -0,0 +1,42 @@
+class EventEmitter {
+   listeners = {};
+     
+   on(eventName, fn) {
+      if(!this.listeners[eventName]) {
+         this.listeners[eventName] = [];
+      } 
+      this.listeners[eventName].push(fn) 
+   }
+        
+   off(eventName, fn) {
+      if (this.listeners[eventName]) {
+         this.listeners[eventName] = this.listeners[eventName].filter((handler) => handler !== fn);
+      }
+   }
+  
+   once(eventName, fn) {
+      const onceWrapper = (...args) => {
+         fn(...args);
+         this.off(eventName, onceWrapper);
+      };
+      this.on(eventName, onceWrapper);
+   }
+  
+   emit(eventName, ...args) {
+      if(this.listeners[eventName]) {
+         this.listeners[eventName].map(fun => args ? fun(...args) : fun())
```

Comment:
You don't need to use the `map` method here, the `forEach` method is more appropriate here. The `map` should be used when you transform the values of the array and want to save them to a new array or return as a result.